### DISCUSSION TITLE
Deployment problem

### DISCUSSION THREAD

**Discussion by m-tranter** [vikejs/vike#1059] [Discussion]
Hi,

I have used the vue-pinia template and got my app running in dev.
How do I serve it using Express? I looked at the boilerplate for this and tried this:
```
      // For HTTP streams use httpResponse.pipe() instead, see https://vite-plugin-ssr.com/stream
      res.send(body)
```
But I don't understand the comment.
I get this error: 
```
Error: [vite-plugin-ssr][Wrong Usage] pageContext.httpResponse.body can't be used because the onRenderHtml() hook defined by /renderer/+onRenderHtml.js > `export default` provides a Readable Node.js Stream. Use `pageContext.httpResponse.pipe()` or `pageContext.httpResponse.getBody()` instead. See https://vite-plugin-ssr.com/stream for more information.
    at getHttpResponseBody (/home/mark/work/vike/vite_rangers/node_modules/vite-plugin-ssr/dist/cjs/node/runtime/renderPage/getHttpResponseBody.js:10:33)
    at get body [as body] (/home/mark/work/vike/vite_rangers/node_modules/vite-plugin-ssr/dist/cjs/node/runtime/renderPage/createHttpResponseObject.js:58:66)
    at /home/mark/work/vike/vite_rangers/server/index.js:56:15
```

**Comment by brillout** [comment]
`body` is synchronous whereas streams are inherenty async. Btw: https://brillout.github.io/rules &mdash; most questions you have are more related to JavaScript and less related to vite-plugin-ssr. I'm sure there are a lot of JavaScript forums out there to ask such questions. (As you can imagine maintaining vite-plugin-ssr is quite a time consuming task.)

**Reply by m-tranter** [reply]
Thank you for directing me to your rules. :D I have spent many hours working with your examples and trying to get it working. What I am trying to do us get an app to work using ssr, so that it will get picked up by search engines. The problem I have is that the CMS I am using doesn't seem to work with a standard Nuxt build. So I wanted something that gave me more control over the build, so that the js could be inlined into the html file. Sorry if your tool isn't the right one for this. Any pointers appreciated.
